dirty_cat/super_vectorizer.py

```diff
@@ -169,45 +168,62 @@ def __init__(self, *,
         self.verbose = verbose
 
     @staticmethod
-    def _auto_cast(X: pd.DataFrame) -> pd.DataFrame:
+    def _auto_cast(X) -> pd.DataFrame:
         """
-        Takes a pandas DataFrame and tries to convert its columns to the best
+        Takes an array-like or a dataframe and tries to convert its columns to the best
         possible data type.
 
         Parameters
         ----------
-        X: pd.DataFrame
-            Input data as a pandas DataFrame.
+        X : {array-like, dataframe} of shape (n_samples, n_features)
+            The data to be transformed.
 
         Returns
         -------
-        array
-            The same array, with its columns casted to the best possible
+        pd.DataFrame
+            The same array as a pandas DataFrame, with its columns casted to the best possible
             data type.
-            Columns with missing values won't be modified.
         """
         from pandas.core.dtypes.base import ExtensionDtype
+        # To use TextParser, we need to have a list of rows
+        colnames = None  # save the column names which we lose during the conversion
+        if isinstance(X, pd.DataFrame):
+            colnames = X.columns
+            X = X.to_numpy().tolist()
```

Comment:
I fear that on large arrays, this will blow the memory. Can we do type detection only on the first few thousand lines?